http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -388,4 +388,35 @@ public static void runAndLogError(NettyClientLogger log, String errorMsg, Functi
             log.error(null, () -> errorMsg, e);
         }
     }
+
+    //  ALPN supported backported in u251
+    //  https://bugs.openjdk.org/browse/JDK-8242894
+    public static boolean isAlpnSupported() {
```

Comment:
What if customers are using openssl, then Java version is no longer a limitation, right?

Instead of checking java version (that seems error-prone), should we check if ALPN related method is there?

Reply:
Updated to check for `ALPNExtension`, copied from Netty

https://netty.io/4.0/xref/io/netty/handler/ssl/JettyAlpnSslEngine.html#44

Reply:
Nvm, this doesn't work, always throws error. Its not clear what classes/methods we need to check, the [old deprecated Netty class](https://github.com/netty/netty/blob/56ef371676ed48262a46ce827d5462cd6f218a6d/handler/src/main/java/io/netty/handler/ssl/JdkAlpnApplicationProtocolNegotiator.java#L29) checked from a handful of sources, the [new class](https://github.com/netty/netty/blob/56ef371676ed48262a46ce827d5462cd6f218a6d/handler/src/main/java/io/netty/handler/ssl/ApplicationProtocolConfig.java) doesn't have checks. Reverting to checking Java version for now, will update

Reply:
This is the check implemented in in Netty 5.x fwiw https://github.com/netty/netty/blob/netty-5.0.0.Alpha5/handler/src/main/java/io/netty5/handler/ssl/JdkAlpnSslUtils.java#L108

Any updates on non JDK Ssl provider? Seems we can skip the check if SslProvider is not JDK
> What if customers are using openssl, then Java version is no longer a limitation, right?

Reply:
+1, I was going to suggest a similar check to This is the check implemented in in Netty 5.x fwiw https://github.com/netty/netty/blob/netty-5.0.0.Alpha5/handler/src/main/java/io/netty5/handler/ssl/JdkAlpnSslUtils.java#L108

Reply:
Updated check to return true ifSslProvider is OpenSsl